mesonbuild/envconfig.py

```diff
@@ -375,8 +376,25 @@ def __init__(
                 if not isinstance(command, (list, str)):
                     raise mesonlib.MesonException(
                         f'Invalid type {command!r} for entry {name!r} in cross file')
+                command = self._rel2abs_path(command)
                 self.binaries[name] = mesonlib.listify(command)
 
+    def _rel2abs_path(self, command: T.Union[str, T.List[str]]) -> T.Union[str, T.List[str]]:
+        if command:
+            if isinstance(command, list):
+                binary = command[0]
+            else:
+                binary = command
```

Comment:
Call `mesonlib.listify` first so you know it's always a list, we listify anyway just after this call.

Reply:
Thanks, first calling `mesonlib.listify` makes this a lot easier.